packages/components/src/popover/index.js

```diff
@@ -42,405 +46,71 @@ const SLOT_NAME = 'Popover';
- *                          attribute.
- */
-function setAttribute( element, name, value ) {
-	if ( ! value ) {
-		if ( element.hasAttribute( name ) ) {
-			element.removeAttribute( name );
-		}
-	} else if ( element.getAttribute( name ) !== value ) {
-		element.setAttribute( name, value );
-	}
-}
-
-/**
- * Sets or removes an element style property.
- *
- * @param {Element} element  The element to modify.
- * @param {string}  property The property to set or remove.
- * @param {?string} value    The value to set. A falsy value will remove the
- *                           property.
- */
-function setStyle( element, property, value = '' ) {
-	if ( element.style[ property ] !== value ) {
-		element.style[ property ] = value;
-	}
-}
-
-/**
- * Sets or removes an element class.
- *
- * @param {Element} element The element to modify.
- * @param {string}  name    The class to set or remove.
- * @param {boolean} toggle  True to set the class, false to remove.
- */
-function setClass( element, name, toggle ) {
-	if ( toggle ) {
-		if ( ! element.classList.contains( name ) ) {
-			element.classList.add( name );
-		}
-	} else if ( element.classList.contains( name ) ) {
-		element.classList.remove( name );
-	}
-}
-
-function getAnchorDocument( anchor ) {
-	if ( ! anchor ) {
-		return;
-	}
-
-	if ( anchor.endContainer ) {
-		return anchor.endContainer.ownerDocument;
-	}
-
-	if ( anchor.top ) {
-		return anchor.top.ownerDocument;
-	}
-
-	return anchor.ownerDocument;
-}
+	return y;
+};
 
 const Popover = (
 	{
+		// Disable reason: We generate the `...contentProps` rest as remainder
+		// of props which aren't explicitly handled by this component.
+		/* eslint-disable no-unused-vars */
+		range,
+		animate,
+		/* eslint-enable no-unused-vars */
 		headerTitle,
 		onClose,
 		children,
 		className,
 		noArrow = true,
 		isAlternate,
-		// Disable reason: We generate the `...contentProps` rest as remainder
-		// of props which aren't explicitly handled by this component.
-		/* eslint-disable no-unused-vars */
-		position = 'bottom right',
-		range,
+		position,
+		placement = 'bottom-start',
+		offset,
 		focusOnMount = 'firstElement',
 		anchorRef,
 		anchorRect,
 		getAnchorRect,
 		expandOnMobile,
-		animate = true,
 		onFocusOutside,
-		__unstableStickyBoundaryElement,
 		__unstableSlotName = SLOT_NAME,
 		__unstableObserveElement,
-		__unstableBoundaryParent,
 		__unstableForcePosition,
-		__unstableForceXAlignment,
-		__unstableEditorCanvasWrapper,
-		/* eslint-enable no-unused-vars */
 		...contentProps
 	},
 	ref
 ) => {
-	const anchorRefFallback = useRef( null );
-	const contentRef = useRef( null );
-	const containerRef = useRef();
+	const arrowRef = useRef( null );
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
-	const [ animateOrigin, setAnimateOrigin ] = useState();
+	const isExpanded = expandOnMobile && isMobileViewport;
+	const hasArrow = ! isExpanded && ! noArrow;
+	const usedPlacement = position
```

Comment:
We should add a deprecation here for `position`.

Reply:
cc https://github.com/WordPress/gutenberg/pull/40740#issuecomment-1117123413